### PR TITLE
Csi vol driver

### PR DESCRIPTION
**Pull request by Rohit-PX**
Why do we need this PR?

Currently, storage provider is evaluated by scheduler driver, moving it to volume driver.

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -28,6 +28,22 @@ import (
...
+	//PortworxStorage portworx storage name
+	PortworxStorage StorageProvisionerType = "portworx"
+	//CsiStorage csi storage name
+	CsiStorage StorageProvisionerType = "csi"
+)
+
+var provisioners = map[StorageProvisionerType]string{
+	PortworxStorage: "kubernetes.io/portworx-volume",
+	CsiStorage:      "com.openstorage.pxd",
+	//CsiStorage: "pxd.portworx.com",
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Remove commented code

File: drivers/volume/portworx/portworx.go [hunk 2]
```diff
@@ -125,6 +141,16 @@ func (d *portworx) Init(sched string, nodeDriver string) error {
 		)
 	}
 
+	logrus.Infof("RK=> %v", storageProvisioner)
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Remove debug log

File: drivers/volume/portworx/portworx.go [hunk 3]
```diff
@@ -125,6 +141,16 @@ func (d *portworx) Init(sched string, nodeDriver string) error {
...
+	if storageProvisioner != "" {
+		if p, ok := provisioners[StorageProvisionerType(storageProvisioner)]; ok {
+			logrus.Infof("RK=> From map  %v", p)
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Remove debug log

File: drivers/volume/portworx/portworx.go [hunk 4]
```diff
@@ -125,6 +141,16 @@ func (d *portworx) Init(sched string, nodeDriver string) error {
...
+			torpedovolume.StorageProvisioner = p
+		}
+	} else {
+		logrus.Infof("RK=> Default storage  %s", string(PortworxStorage))
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Remove debug log

File: drivers/volume/portworx/portworx.go [hunk 5]
```diff
@@ -28,6 +28,22 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+// StorageProvisionerType provisioner to be used for torpedo volumes
+type StorageProvisionerType string
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go]: Move this to `volume.go`

File: drivers/volume/volume.go [hunk 6]
```diff
@@ -136,8 +136,13 @@ type Driver interface {
 
 var (
 	volDrivers = make(map[string]Driver)
+	// StorageProvisioner to be used to store name of the storage provisioner
+	StorageProvisioner string
```
Comment (disrani-px) [inline drivers/volume/volume.go]: Change the type to `StorageProvisionerType`

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (Rohit-PX) [reply drivers/volume/portworx/portworx.go]: Fixed.

File: drivers/volume/volume.go (hunk 6 above)
Comment (Rohit-PX) [reply drivers/volume/volume.go]: Fixed.

File: drivers/volume/portworx/portworx.go (hunk 4 above)
Comment (Rohit-PX) [reply drivers/volume/portworx/portworx.go]: Fixed.

File: drivers/volume/portworx/portworx.go (hunk 3 above)
Comment (Rohit-PX) [reply drivers/volume/portworx/portworx.go]: Fixed.

File: drivers/volume/portworx/portworx.go (hunk 2 above)
Comment (Rohit-PX) [reply drivers/volume/portworx/portworx.go]: Fixed.

**Pull request merged**